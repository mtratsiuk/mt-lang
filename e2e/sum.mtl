(def sum
  (|a b c|
    (+
      a
      (+
        b
        c))))

(print (sum 10 20 30))
(print ((sum 10) 20 30))
(print (((sum 10) 20) 30))
