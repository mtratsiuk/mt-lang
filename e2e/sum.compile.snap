const __mtl_curry = f => (...args) => {
  if (args.length >= f.length) {
    return f(...args);
  }

  return __mtl_curry(f.bind(null, ...args));
};

const sum = __mtl_curry((a, b, c) => {
  return (a + (b + c));
});

console.log(String(sum(10, 20, 30)));

console.log(String(sum(10)(20, 30)));

console.log(String(sum(10)(20)(30)));
